pkg/cpio/cpio.go

```diff
@@ -77,9 +78,28 @@ func (r Reader) ReadRecords() ([]Record, error) {
 
 type Writer struct {
 	rw RecordWriter
+	// There seems to be no harm done in stripping
+	// duplicate names when the record is written,
+	// and lots of harm done if we don't do it.
+	alreadyWritten map[string] bool
 }
 
 func (w Writer) WriteRecord(rec Record) error {
+	// we do NOT write records with absolute paths.
+	if filepath.IsAbs(rec.Name) {
```

Comment:
yep, you're right. But unlike the linux world we have lots of cpio generators because we have a package. And any cpio generator that creates absolute paths is creating a time bomb, which will detonate on the first person to do this as root
cpio -it < /whatever.cpio
because they forgot to type -ivt.

they'll trash their machine. And they'll hate us forever. 

You might wonder how I know this. It's because I've done it. And we had people do it last year (not with u-root, with something else). I really don't like letting absolute paths into archives. I'd prefer to leave this here instead of requiring every possible user of our package to not make that mistake. There is no justification for absolute paths in archives. Ever.

I realize this does not belong here. But, that said, I think it's the right thing to do. I don't like the consequences of not doing it. Also, if you don't do it here, you've got to remember to do it in every tool that generates cpio, which is really nasty and awful. I went with the first option (each tool does it) and it was unpleasant.